tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -54,6 +57,26 @@ protected void addBadWords() {
 	}
 
 	@Override
+	protected void genFile(Grammar g, ST outputFileST, String fileName) {
+		super.genFile(g, outputFileST, fileName);
+		gofmt(fileName); // consider commenting out when debugging
+	}
+
+	private void gofmt(String fileName) {
+		// If the generated source file is for any reason invalid, gofmt will not change the file and generate output on
```

Comment:
We do generate invalid code at the moment. Hopefully we will stop doing that soon. :D